cloudify_marketplace/blueprint/scripts/configure_aws.py

```diff
@@ -0,0 +1,168 @@
+import os
+import urllib2
+from ConfigParser import ConfigParser
+
+from cloudify import ctx
+from cloudify_rest_client import CloudifyClient
+from cloudify.state import ctx_parameters as inputs
+
+
+BOTO_CONF = os.path.expanduser('~/.boto')
+
+
+def get_manager_instance(conn):
+    url = 'http://169.254.169.254/latest/meta-data/instance-id'
+    instance_id = urllib2.urlopen(url).read().split()[0]
+
+    return conn.get_only_instances(instance_ids=[instance_id])[0]
+
+
+def get_region():
+    url = 'http://169.254.169.254/latest/meta-data/placement/availability-zone'
+    return urllib2.urlopen(url).read()[:-1]
+
+
+def create_boto_config(path, aws_access_key_id, aws_secret_access_key, region):
+    config = ConfigParser()
+
+    config.add_section('Credentials')
+    config.set('Credentials',
+               'aws_access_key_id',
+               aws_access_key_id)
+    config.set('Credentials',
+               'aws_secret_access_key',
+               aws_secret_access_key)
+
+    config.add_section('Boto')
+    config.set('Boto', 'ec2_region_name', region)
+
+    ctx.logger.info('Saving boto config: {0}'.format(path))
+    with open(path, 'w') as fh:
+        config.write(fh)
+
+
+def create_agents_security_group(conn,
+                                 agents_sg_name,
+                                 manager_security_group,
+                                 vpc_id):
+    ctx.logger.info('Creating agent security group: {0}'
+                    .format(agents_sg_name))
+
+    agents_sg_desc = 'Security group for Cloudify agent VMs'
+    agents_sg = conn.create_security_group(
+        agents_sg_name,
+        agents_sg_desc,
+        vpc_id=vpc_id,
+    )
+
+    # authorize from manager to agents
+    add_tcp_allows_to_security_group(
+        port_list=[22, 5985],
+        security_group=agents_sg,
+        from_group=manager_security_group,
+    )
+
+    return agents_sg
+
+
+def configure_manager_security_group(manager_security_group,
+                                     agents_security_group):
+    # authorize from agent to manager
+    add_tcp_allows_to_security_group(
+        port_list=[5672, 8101, 53229],
```

Comment:
I'll move it to an input on the blueprints (which is where it probably should be anyway)